crates/exex/src/hints.rs

```diff
@@ -112,3 +115,152 @@ pub fn add_segment_hint() -> Hint {
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.mix_hash.0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.mix_hash.0[0..16],
+            )));
+            block_header
+                .push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(&header.nonce.0)));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.number.to_be_bytes::<{ U256::BYTES }>(),
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.parent_beacon_block_root.unwrap().0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.parent_beacon_block_root.unwrap().0[0..16],
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.parent_hash.0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.parent_hash.0[0..16],
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.receipt_trie.0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.receipt_trie.0[0..16],
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.state_root.0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.state_root.0[0..16],
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.timestamp.to_be_bytes::<{ U256::BYTES }>(),
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.transactions_trie.0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.transactions_trie.0[0..16],
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.uncle_hash.0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.uncle_hash.0[0..16],
+            )));
+
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.withdrawals_root.unwrap().0[16..],
+            )));
+            block_header.push(MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                &header.withdrawals_root.unwrap().0[0..16],
+            )));
+
```

Comment:
I'd like us to work on types, so I'd like to have a. `block.into()` here in the hints and all required conversion in a dedicated trait

Reply:
we have a `trait KethBlockHeader`, `KethBlock` that implements `from` for various input types as you mentioned these ones and the ef tests ones

Reply:
Just added the `pub trait KethBlock` trait to handle the three conversions:
- From reth header to a vec of MaybeRelocatable that can be understood by the cairo vm
- From reth block body to a vec of MaybeRelocatable that can be understood by the cairo vm
- Full conversion that includes Header + Body